### ISSUE TITLE
Row / Column Styling Issues

### ISSUE THREAD

**Issue by JatinChimote** [iliaal/php_excel#123]
Using latest Php_excel and Libxl 3.6.1.1 (trial):

=> Not sure if this is a Bug in php_excel or libxl or a limitation of the trial Version I am using, or something wrong I am doing please help

If I took a xlsx file and styled either entire row or the entire column like the fig a) and fig b) respectively.

Fig a)
![row](https://cloud.githubusercontent.com/assets/2509409/6666172/c6512748-cc07-11e4-90d0-8bc42899fbc8.png)

Fig b) 
![column](https://cloud.githubusercontent.com/assets/2509409/6666176/d0fac0e6-cc07-11e4-9b96-fbb10ab24409.png)

Note: Done one thing at a time either column formatting or row formatting.

Now, if you write rows by loading this xslx. with code like the given code.

```php
$filePath = './pivottest.csv';

include('csv2array.php');

// Specify Dummy License 
$doc = new ExcelBook("AAA","AAA", true);

//Sample row 
// $row = array(1, 'Carrots', 'Vegetables', 15000, '1/6/2012', 'United States');

//$row = $data[0];  # Use when using write (cell)

// Load the template 
$doc->loadFile("Simple.xlsx");

// Set the reference to a specific Sheet 
$sheet = $doc->getSheet(0); 

// Write to Some Cells 
//$sheet->write(2,3, $row[3]);

foreach ($data as $rowno => $row) { 
	$row[0] = (int) $row[0];  // Convert to int to avoid the "Number Stored as Text issue"
	$row[3] = (int) $row[3];  // Convert to int to avoid the "Number Stored as Text issue"
	$sheet->writeRow($rowno + 2, $row); // leave 2 rows one for the lixl and one for heading 
}


// Set the header content for the output
header("Content-Type: application/octet-stream");
header("Content-Type: application/download");
header("Content-Description: File Transfer");
header('Content-Type: text/xlsx; charset=utf-8');
header('Content-Disposition:attachment;filename=output.xlsx');

// Write as another file.
echo $doc->save();
``` 
The output is that the most values in the in the row do not get any formatting while some cells (only one in my case) do:

![result](https://cloud.githubusercontent.com/assets/2509409/6667146/208655fa-cc10-11e4-8221-3fee39aab6fa.png)

However, If I select individual cells or range then do the styling it works without issues.
Any idea's why ?

Update:
Also seems that this is also with Formatting the rows or Columns i.e formatting is lost when you select column and right click and format (to say Currency), same as done for styling above but if you apply selectively apply this to some cells the formatting works. Both these issues seems to be somehow interlinked also.

**Comment by johmue** [comment]
It seems to be not related to trial/commercial since it behaves the same way on both modes. I figured out that it makes a difference if you select a fixed range of cells or if you select a row/column as a whole. It seems to be also affected by the type of the value that is written.

At this time I'm not sure if this is a LibXL problem or a problem related to this project. I will investigate and hopefully come up with an answer soon.

**Comment by johmue** [comment]
You should consider to use the below approach of reading the cell format first and then write the cell value back with the given format.

```php
// ...
// iterate on rows
foreach ($data as $rowno => $row) {
    // iterate on columns
    foreach ($row as $colno => $cellVal) {
        // read format for specific cell
        $format = $sheet->cellFormat($rowno + 2, $colno);

        // write cell value with specific format
        $sheet->write($rowno + 2, $colno, $cellVal, $format);  
    }
}
// ...
```

The issue itself is most likely related to the implementation of ExcelSheet::writeRow() in the extension but "fixing" it would slow down the extension since you would use the exact same approach as above but on C level. One might consider to add an indicator to writeRow() to handle this kind of edge cases.

**Comment by johmue** [comment]
...might be fixed with libxl 3.6.3+ according changelog...

**Comment by iliaal** [comment]
Fixed in latest libxl as far as I can tell
